infrastructure/README.md

```diff
@@ -131,3 +131,51 @@ KMD is not started by default (with `make start-private`). You need to run it se
 2. `remove-reach`: halts any dockerized devnets, kills & removes docker instances and containers, remove reach bash file from `/home/user/algorand-reach`.
 
 After setup, if you want to use reach commands directly you can use `~/algorand-reach/reach <command>`. For eg, after `setup-reach`, try running `~/algorand-reach/reach --help` to see a list of available commands.
+
+## Indexer v2
+
+The [Indexer](https://developer.algorand.org/articles/introducing-algorands-v2-indexer/) is a standalone service that reads committed blocks from the Algorand blockchain and maintains a database of transactions and accounts that are searchable and indexed.
+
+By default the indexer runs on port `8980` and `postgresdb` on port `5432` (on local or either in a docker container).
+
+### Installation with docker
+**Prerequisite:** Make sure to have [Docker Compose](https://docs.docker.com/compose/install/) installed (with non root privilages).
+
+Following make jobs are provided:
+1. `indexer-docker-up`: Clones the indexer repo to `~/.algorand-indexer` and runs `docker-compose up` on the [docker-compose.yml](https://github.com/algorand/indexer/blob/develop/docker-compose.yml) file.
+2. `indexer-docker-down`: Stops and removes indexer related container and images.
+
+**NOTE:** Docker based setup runs indexer in a "[read-only](https://github.com/algorand/indexer/blob/develop/docker/run.sh#L10)" mode, without connecting to the private-net `algod` node. Read more about this mode [here](https://github.com/algorand/indexer#read-only).
+
+### Installation on local
+
+The Indexer primarily provides two services: loading a PostgreSQL database with ledger data and supplying a REST API to search this ledger data. For this purpose, setting up indexer on local consists of two steps:
+1.  `make setup-postgresql`: Setting up [`postgresql`](https://www.postgresql.org/) database on your local machine.
+2. `make setup-indexer`: Start indexer by connecting to local-db and your private-net algod node.
+
+#### Step1: Setup postgres-db
+
+For this we provide `make setup-postgresql` command. Running it will [install](https://www.postgresql.org/download/linux/ubuntu/) postgresql database on your system and setup a new user & database. Config vars are provided below:
+```
+host=localhost
+port=5432
+username=algorand
+dbname=pgdb
+password=indexer
+```
+
+Please note that this installation is currently compatible with `linux based` distributions only. For other OS, try the installation from [source](https://www.postgresql.org/download/).
+
+#### Step2: Start Indexer
+
+**Note:** Make sure you have setup a private network up and running (using `make setup-private-net`). For more instructions check [setting up a local node](./#setting-up-a-local-node) section.
+
+After setting up the database, you can use `make setup-indexer` to add local indexer binary (downloaded in `~/indexer`) and start the indexer by connecting to database and your local algod node.
```

Comment:
aren't we using `.algorand-indexer`? If not, shall we use a directory with algorand in name? Eg: ~/.algorand-indexer-download

Reply:
`~/.algorand-indexer-download` works. Updating.